parser/src/class/parser.rs

```diff
@@ -0,0 +1,111 @@
+use flora_syntax::ClassDeclaration;
+use nom::types::CompleteStr as Input;
+use nom::*;
+
+named!(
+    pub class_declaration<Input, ClassDeclaration>,
+    do_parse!(
+        multispace0 >>
+        tag!("class") >>
+        multispace1 >>
+        name: map!(
+            alphanumeric1,
+            |Input(name)| name
+        ) >>
+        multispace0 >>
+        char!(';') >>
+        multispace0 >>
+        (ClassDeclaration { name })
+    )
+);
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn single_character_name() {
+        assert_eq!(
+            class_declaration(Input("class A;")),
+            Ok((Input(""), ClassDeclaration { name: "A" }))
+        )
```

Comment:
Why is the second argument `Ok((Input(""), ClassDeclaration { name: "A" }))` and not just `ClassDeclaration { name: "A" }))`?